src/components/structures/MatrixChat.js

```diff
@@ -126,6 +143,33 @@ module.exports = React.createClass({
         }
     },
 
+    _registerAsGuest: function() {
+        var self = this;
+        var config = this.props.config;
+        console.log("Doing guest login on %s", config.default_hs_url);
+        MatrixClientPeg.replaceUsingUrls(
+            config.default_hs_url, config.default_is_url
+        );
+        MatrixClientPeg.get().registerGuest().done(function(creds) {
```

Comment:
OK, this is probably fair enough as I think the JS SDK looks the same in either case, as does the interface to an app that uses react-sdk, so the worst that will happen is we or someone else implements an HS that supports UA auth for guests and react-sdk doesn't work with it.